dash-pipeline/SAI/templates/saiapi.h.j2

```diff
@@ -381,6 +381,33 @@ typedef sai_status_t (*sai_bulk_remove_{{ table.name }}_fn)(
         _In_ sai_bulk_op_error_mode_t mode,
         _Out_ sai_status_t *object_statuses);
 
+{% endif %}
+{% if table.enable_query == 'true' %}
+/**
+ * @brief Bulk query {{ sai_api.app_name }}_{{ table.name }}
+ *
+ * @param[in] object_count Number of objects to create
+ * @param[in] {{ table.name }} List of object to create
+ * @param[in] attr_count List of attr_count. Caller passes the number
+ *    of attribute for each object to create.
+ * @param[in] attr_list List of attributes for every object.
+ * @param[in] mode Bulk operation error handling mode.
+ * @param[out] object_statuses List of status for every object. Caller needs to
+ * allocate the buffer
+ *
+ * @return #SAI_STATUS_SUCCESS on success when all objects are created or
+ * #SAI_STATUS_FAILURE when any of the objects fails to create. When there is
+ * failure, Caller is expected to go through the list of returned statuses to
+ * find out which fails and which succeeds.
+ */
+typedef sai_status_t (*sai_bulk_query_{{ table.name }}_fn)(
+        _In_ uint32_t object_count,
+        _In_ const sai_{{ table.name }}_t *{{ table.name }},
+        _In_ const uint32_t *attr_count,
+        _In_ const sai_attribute_t **attr_list,
+        _In_ sai_bulk_op_error_mode_t mode,
+        _Out_ sai_status_t *object_statuses);
```

Comment:
I believe we still need the inout attributes for handling the direct query case.